### PR TITLE
Enhance the error message for integration installation

### PR DESCRIPTION
**Pull request by nabenabe0928**
## Motivation

This PR is a followup of this PR:
- https://github.com/optuna/optuna-integration/pull/130

## Description of the changes

Enhance the error message for `from optuna.integration import X` and `from optuna.integration.Y import X`.

> [!NOTE]
> We do not have to use `try_import` in each integration module file because `__init__.py` does the lazy import for each of them.
> However, `ImportError` in each integration module file is necessary as we would like to raise an error when `optuna.integration.X` path is specified explicitly.

### CODE REVIEW

**Comment by eukaryo** [comment]
@gen740 @not522 Could you review this PR?

File: optuna/integration/__init__.py [hunk 1]
```diff
@@ -129,12 +131,15 @@ def _get_module(self, module_name: str) -> ModuleType:
...
                 return importlib.import_module("." + module_name, self.__name__)
             except ModuleNotFoundError:
-                raise ModuleNotFoundError(
-                    "Optuna's integration modules for third-party libraries have started "
-                    "migrating from Optuna itself to a package called `optuna-integration`. "
-                    "The module you are trying to use has already been migrated to "
-                    "`optuna-integration`. Please install the package by running "
-                    "`pip install optuna-integration`."
-                )
+                is_module_deprecated = module_name in ("allennlp", "chainer", "chainermn")
```
Comment (not522) [inline optuna/integration/__init__.py:134]: How about adding all the options to `pyproject.toml` in `optuna-integration`? I think no exceptions would lower management costs.

Comment (nabenabe0928) [reply optuna/integration/__init__.py]: I applied your suggestion along with `optuna-integration`:
- https://github.com/optuna/optuna-integration/pull/130

File: optuna/integration/lightgbm.py [hunk 2]
```diff
@@ -28,6 +37,7 @@ class _LightGBMModule(ModuleType):
     __path__ = [os.path.dirname(__file__)]
 
     def __getattr__(self, name: str) -> Any:
+        print(lgb.__dict__)
```
Comment (not522) [inline optuna/integration/lightgbm.py:40]: It seems you forgot to delete it.

Comment (nabenabe0928) [reply optuna/integration/lightgbm.py]: Good catch, I deleted this line!

**Review by not522 — APPROVED** [review]
LGTM!

**Comment by nabenabe0928** [comment]
@not522 
Could you also review this PR (or suggest another reviewer)?
- https://github.com/optuna/optuna-integration/pull/130

The PR above is a blocker of this PR.

**Review by gen740 — APPROVED** [review]
LGTM!

**Comment by gen740** [comment]
@nabenabe0928 Is it alright to merge this PR?

**Comment by nabenabe0928** [comment]
@gen740 yes!

**Comment by not522** [comment]
I'll merge this PR after https://github.com/optuna/optuna-integration/pull/130 is merged.

**Comment by HideakiImamura** [comment]
[FYI] https://github.com/optuna/optuna-integration/pull/130 has been merged.

**Pull request merged**